src/utils/preRestart.ts

```diff
@@ -1,28 +1,45 @@
 /* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Signer } from 'ethers'
 import { Route } from '../types'
 import { LifiErrorCode } from './errors'
 
-export const handlePreRestart = (route: Route) => {
+export const handlePreRestart = async (route: Route, signer: Signer) => {
   for (let index = 0; index < route.steps.length; index++) {
     const stepHasFailed = route.steps[index].execution?.status === 'FAILED'
 
     if (stepHasFailed) {
-      handleErrorType(route, index)
+      await handleErrorType(route, index, signer)
       deleteFailedProcesses(route, index)
       deleteTransactionData(route, index)
     }
   }
 }
 
-const handleErrorType = (route: Route, index: number) => {
+const handleErrorType = async (route: Route, index: number, signer: Signer) => {
   const isGasLimitError = route.steps[index].execution?.process.some(
     (p) => p.error?.code === LifiErrorCode.GasLimitError
   )
   const isGasPriceError = route.steps[index].execution?.process.some(
     (p) => p.error?.code === LifiErrorCode.TransactionUnderpriced
   )
 
+  const { transactionRequest } = route.steps[index]
+
   if (isGasLimitError) {
+    if (transactionRequest) {
+      let gasLimit = transactionRequest.gasLimit
+
+      try {
+        gasLimit = await signer.estimateGas(transactionRequest)
+      } catch (error) {}
+
+      if (gasLimit) {
+        transactionRequest.gasLimit = `${
+          (BigInt(gasLimit?.toString()) * 125n) / 100n
```

Comment:
Unnecessary optional chaining here due to the above check. Also, passing undefined to BigInt will throw TypeError.